4ws:

  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100
  

  # Velocity Controllers ---------------------------------------
  fl_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: fl_wheel_joint
    #pid: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  fr_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: fr_wheel_joint
    #pid: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  rl_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: rl_wheel_joint
    #pid: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  rr_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: rr_wheel_joint
    #pid: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}


  # Position Controllers ---------------------------------------
  fl_steering_position_controller:
    joint: fl_steering_joint
    type: position_controllers/JointPositionController
    #pid: {p: 10.0, i: 0.0, d: 0.5}
  fr_steering_position_controller:
    joint: fr_steering_joint
    type: position_controllers/JointPositionController
    #pid: {p: 10.0, i: 0.0, d: 0.5}
  rl_steering_position_controller:
    joint: rl_steering_joint
    type: position_controllers/JointPositionController
    #pid: {p: 10.0, i: 0.0, d: 0.5}
  rr_steering_position_controller:
    joint: rr_steering_joint
    type: position_controllers/JointPositionController
    #pid: {p: 10.0, i: 0.0, d: 0.5}

#4ws/gazebo_ros_control/pid_gains:
  #fr_wheel_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  #fl_wheel_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  #rr_wheel_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  #rl_wheel_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp: 0.0}
  #fr_steering_joint: {p: 8.5, i: 0.0, d: 5.3}
  #fl_steering_joint: {p: 8.5, i: 0.0, d: 5.3}
  #rr_steering_joint: {p: 8.5, i: 0.0, d: 5.3}
  #rl_steering_joint: {p: 8.5, i: 0.0, d: 5.3}
  
